store/postgres/src/detail.rs

```diff
@@ -277,25 +273,57 @@ pub(crate) fn deployment_statuses(
     use subgraph_deployment as d;
     use subgraph_error as e;
 
-    // Empty deployments means 'all of them'
-    if sites.is_empty() {
-        d::table
-            .left_outer_join(e::table.on(d::fatal_error.eq(e::id.nullable())))
-            .load::<(DeploymentDetail, Option<ErrorDetail>)>(conn)?
-            .into_iter()
-            .map(|(detail, error)| status::Info::try_from(DetailAndError(detail, error, sites)))
-            .collect()
-    } else {
-        let ids: Vec<_> = sites.into_iter().map(|site| site.id).collect();
+    // First, we fetch all deployment information along with any fatal errors.
+    // Subsequently, we fetch non-fatal errors and we group them by deployment
+    // ID.
+
+    let details_with_fatal_error = {
+        let join = e::table.on(e::id
+            .nullable()
+            .eq(d::fatal_error)
+            .and(e::subgraph_id.eq(d::deployment)));
```

Comment:
This `and` cause is redundant, the deployment id is hashed into the error id, so if the error id is equal the deployment id must also be equal.

Reply:
Fixed by https://github.com/graphprotocol/graph-node/pull/3350/commits/161f78997d689767eeaa60af3c6ed1d911c60920.